astropy/io/fits/column.py

```diff
@@ -1423,7 +1423,10 @@ def dtype(self):
                 else:
                     dt = np.dtype((dt.base, dim))
 
-            fields.append((name, dt))
+            if PY2:  # pragma: py2
+                fields.append((name.encode('utf-8'), dt))
```

Comment:
I don't know how the string `.encode()` method is written but hopefully it is smart enough not to. Existing tests are not complaining.